datumaro/components/explorer.py

```diff
@@ -43,22 +43,25 @@ def __init__(
         """
         self._model = ExplorerLauncher(model_name="clip_visual_ViT-B_32")
         self._text_model = ExplorerLauncher(model_name="clip_text_ViT-B_32")
-        inference = dataset.run_model(self._model, append_annotation=True)
         self._topk = topk
-
         database_keys = []
         item_list = []
-
-        for item in inference:
-            for annotation in item.annotations:
-                if isinstance(annotation, HashKey):
-                    try:
-                        hash_key = annotation.hash_key[0]
-                        hash_key = np.unpackbits(hash_key, axis=-1)
-                        database_keys.append(hash_key)
-                        item_list.append(item)
-                    except Exception:
-                        hash_key = None
+        if not isinstance(dataset, List):
```

Comment:
How do you think about below suggestion if the dataset argument incomes only as `List[IDataset]` or `IDataset`.
```suggestion
        if isinstance(dataset, IDataset):
```